build.sh

```diff
@@ -76,6 +76,38 @@ function clean_image {
   fi
 }
 
+# Pull image based on FROM, before we build our own.
+function pull_image {
+  local dockerfile="$1"
+
+  # Get image_name from Dockerfile before pulling.
+  while read -r line; do
+    if ! grep -q "^FROM" <<< "$line"; then
+      continue
+    fi
+
+    # In case FROM scratch is defined, skip it
+    if grep -q "scratch" <<< "$image_name"; then
```

Comment:
Just a nitpick, please check that the image name is exactly `scratch`, a correct image could in theory have scratch in its name.

Reply:
Also, the assignment to `image_name` should be before this line

Reply:
Addressed.